tests/test_butterworth_filter.py

```diff
@@ -0,0 +1,192 @@
+"""Tests for the Butterworth time domain filter"""
+import numpy as np
+import pytest
+from scipy import signal
+from eitprocessing.filters.butterworth_filters import BandPassFilter
+from eitprocessing.filters.butterworth_filters import BandStopFilter
+from eitprocessing.filters.butterworth_filters import ButterworthFilter
+from eitprocessing.filters.butterworth_filters import HighPassFilter
+from eitprocessing.filters.butterworth_filters import LowPassFilter
+
+
+@pytest.fixture
+def filter_arguments():
+    return {
+        "filter_type": "lowpass",
+        "cutoff_frequency": 10,
+        "sample_frequency": 50.2,
+        "order": 5,
+    }
+
+
+def check_filter_attributes(filter_, kwargs):
+    for key, value in kwargs.items():
+        assert getattr(filter_, key) == value
+
+    assert hasattr(filter_, "apply_filter")
+    assert callable(filter_.apply_filter)
+    assert filter_.available_in_gui
+
+
+def test_create_butterworth_filter(filter_arguments):
+    filter_ = ButterworthFilter(**filter_arguments)
+    check_filter_attributes(filter_, filter_arguments)
+
+
+def test_butterworth_range(filter_arguments):
+    # Tests whether an out-of-range order raises AttributeError
+    del filter_arguments["order"]
+
+    with pytest.raises(ValueError):
+        ButterworthFilter(**filter_arguments, order=11)
+
+    # Tests whether an explicit out-of-range order does not raise AttributeError
+    try:
+        filter_ = ButterworthFilter(**filter_arguments, order=11, override_order=True)
+    except AttributeError:
+        pytest.fail("Unexpected AttributeError")
+    assert filter_.order == 11
+
+
+def test_butterworth_filter_type(filter_arguments):
+    filter_arguments["filter_type"] = "invalid"
+    with pytest.raises(ValueError):
+        ButterworthFilter(**filter_arguments)
+
+
+def test_butterworth_cutoff_frequency(filter_arguments):
+    del filter_arguments["cutoff_frequency"]
+
+    with pytest.raises(TypeError):
+        ButterworthFilter(**filter_arguments, cutoff_frequency="not a number")
+
+    filter_arguments["filter_type"] = "bandpass"
+    with pytest.raises(TypeError):
+        ButterworthFilter(**filter_arguments, cutoff_frequency="not a number")
+
+    with pytest.raises(TypeError):
+        ButterworthFilter(**filter_arguments, cutoff_frequency=10)
+
+    with pytest.raises(TypeError):
+        ButterworthFilter(**filter_arguments, cutoff_frequency=("not a number", True))
+
+    with pytest.raises(TypeError):
+        ButterworthFilter(**filter_arguments, cutoff_frequency=[20, 30])
+
+    with pytest.raises(ValueError):
+        ButterworthFilter(**filter_arguments, cutoff_frequency=(1,))
+
+    with pytest.raises(ValueError):
+        ButterworthFilter(**filter_arguments, cutoff_frequency=(1, 2, 3))
```

Comment:
I think it would be clearer (and easier to add other options if we think of any) to loop over some invalid options like this.

```suggestion
    filter_arguments["filter_type"] = "bandpass"
    invalid_bandpass_cutoffs = [
        "not a number", 
        10, 
        ("not a number", True),
        [20,30],
        (1,)
        (1,2,3),
        ]
    for invalid in invalid_bandpass_cutoffs:
        with pytest.raises(TypeError):
            ButterworthFilter(**filter_arguments, cutoff_frequency=invalid)
```

Reply:
Thank you for the suggestion. I've implemented it in 29a6d580577527ed5a7bd0320722b222e3a0c982.